zio-http/src/main/scala/zio/http/URL.scala

```diff
@@ -25,10 +25,19 @@ final case class URL(
 
   def encode: String = URL.encode(self)
 
-  def host: Option[String] = kind match {
+  def host: Option[String]                 = kind match {
     case URL.Location.Relative      => None
     case abs: URL.Location.Absolute => Option(abs.host)
   }
+  def hostWithOptionalPort: Option[String] =
```

Comment:
Maybe this is telling us that `host` should return a `Host` that has `hostname` and `port`?

Reply:
We already have `URL.Location.Absolute(scheme, host, port)` which is similar.. Not sure. Can we open a separate ticket for possibly redesigning the `URL` model?

Reply:
Sounds good.